### PR TITLE
Add optional wdir to exec()

### PR DESCRIPTION
**Pull request by gongsu832**
Add optional wdir to exec() to allow running a command
in a specified work directory. This is used to get around
the problem that old ar command does not support the
--output option.

Signed-off-by: Gong Su <gong_su@hotmail.com>

### CODE REVIEW

File: src/Compiler/CompilerUtils.cpp [hunk 1]
```diff
@@ -239,29 +239,49 @@ struct Command {
...
 
+    // If a work directory is specified, save the current work directory
+    // and switch into it. Note that if wdir is empty, new_wdir will be
+    // cur_wdir.
+    SmallString<8> cur_wdir;
+    SmallString<8> new_wdir(wdir);
+    llvm::sys::fs::current_path(cur_wdir);
+    llvm::sys::fs::make_absolute(cur_wdir, new_wdir);
+    if (std::error_code ec = llvm::sys::fs::set_current_path(new_wdir)) {
+      cerr << StringRef(new_wdir).str() << ": " << ec.message() << endl;
```
Comment (etiotto) [inline src/Compiler/CompilerUtils.cpp:261]: Suggest to use `llvm::errs()`, I see is used to emit other error messages..

File: src/Compiler/CompilerUtils.cpp [hunk 2]
```diff
@@ -239,29 +239,49 @@ struct Command {
...
     int rc = llvm::sys::ExecuteAndWait(_path, llvm::makeArrayRef(argsRef),
         /*Env=*/None, /*Redirects=*/None,
         /*SecondsToWait=*/0, /*MemoryLimit=*/0, &errMsg);
 
     if (rc != 0) {
-      fprintf(stderr, "%s\n", llvm::join(argsRef, " ").c_str());
-      fprintf(stderr, "Error message: %s\n", errMsg.c_str());
-      fprintf(stderr, "Program path: %s\n", _path.c_str());
-      fprintf(stderr, "Command execution failed.");
+      cerr << llvm::join(argsRef, " ") << endl
```
Comment (etiotto) [inline src/Compiler/CompilerUtils.cpp:276]: Suggest to use `llvm::errs()`, I see is used to emit other error messages..

File: src/Compiler/CompilerUtils.cpp (hunk 1 above)
Comment (AlexandreEichenberger) [reply src/Compiler/CompilerUtils.cpp]: done, thanks Gong.

File: src/Compiler/CompilerUtils.cpp (hunk 2 above)
Comment (AlexandreEichenberger) [reply src/Compiler/CompilerUtils.cpp]: same here

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, @etiotto I believe all your requests are addressed.

**Comment by etiotto** [comment]
Yup thanks @AlexandreEichenberger LGTM too.

**Pull request merged**